src/finitio/support/constraint.coffee

```diff
@@ -40,13 +40,33 @@ class Constraint.Native extends Constraint
 class Constraint.Regexp extends Constraint
   kind: 'regexp'
 
-  accept: (arg) ->
+  accept: (arg, world) ->
     @native.test(arg)
 
+class Constraint.Function extends Constraint
+  kind: 'function'
+
+  accept: (arg, world) ->
+    identifiers = @native.split('.')
+    path = ""
+    func = world
+
+    $u.each identifiers, (id, idx) ->
+      path += '.' if idx > 0
+      path += id
+      func = func[id]
+      unless func
+        throw new Error("#{path} is undefined");
+
+    func(arg)
```

Comment:
Please refactor towards a functional programming style, e.g. folr/reduce or something like that